src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs

```diff
@@ -8,16 +8,19 @@ namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
     using Markdig.Syntax;
     using System;
     using System.Collections.Generic;
+    using System.Diagnostics;
     using System.Net;
     using System.Security.Cryptography;
     using System.Text;
+    using System.Text.RegularExpressions;
 
     public class ImageExtension : ITripleColonExtensionInfo
     {
         public string Name => "image";
         public bool SelfClosing => true;
-        public bool EndingTripleColons => true;
+        public bool EndingTripleColons { get; set; } = false;
```

Comment:
In docfx v3, there is only one global markdown engine, thus `ImageExtension` is also a global singleton that are used by multiple threads concurrently.

Changing this to a settable property leaks internal state across markdown documents and produces wrong result.

Reply:
`EndingTripleColons` flag is consumed in`Parser` stage, (before `Render`), changing the value in `Render` stage may not give you desired behavior

Reply:
Thank you for the insight @yufeih ! I'll work on it!

Reply:
@yufeih -Want to take a look again?

Reply:
This looks correct now